### ISSUE TITLE
Set Environment Variables

### ISSUE THREAD

**Issue by GoogleCodeExporter** [eteran/edb-debugger#14]
I would it if the Environment Viewer plugin could set the environment variables it finds.

**Comment by eteran** [comment]
So, I think the way to do this most "correctly" would be do the following:

1. allocate a new page in the foreign process, probably by forcing a `mmap` call
2. copy the existing environment to that page
3. modify as needed in the new copy
4. set the libc's environment pointer to that new page

Now here's the real question: We know how to do all of the above, but have not had any particular demand for this feature. Do we need an issue for this given that if needed, we know how to do it?

**Comment by 10110111** [comment]
> set the libc's environment pointer to that new page

What if the app isn't using libc, like fasm? Although fasm doesn't check environment, a similar app can still use it.

**Comment by eteran** [comment]
Good point, we'd probably want to find some pointer which is the most "fundamental" to work with then

**Comment by AaronOpfer** [comment]
To execute the child don't we fork and exec? Can't we call setenv before we
exec? I believe we already do this for some environment variables.

**Comment by eteran** [comment]
@AaronOpfer yes, we do and that will cover setting environment variables for a process we are about to start. When I first wrote this FR, I was thinking more along the lines of changing the environment of a process that is already running.
